Provider crashes when custom domain is verified

The API doesn't return the verification field when the custom domain is verified.

```json
[
  {
    "custom_domain_id": "123",
    "domain": "my_domain",
    "primary": true,
    "status": "ready",
    "type": "auth0_managed_certs"
  }
]
```

When the custom domain is verified, `c.Verification.Methods` doesn't exist and the provider crashes.
Affected line: https://github.com/yieldr/terraform-provider-auth0/blob/5cae09177f62e6c6d524f227ce19b51c1a8000e3/auth0/resource_auth0_custom_domain.go#L87

Error in TF:

```
panic: runtime error: invalid memory address or nil pointer dereference
2018-11-04T17:53:00.507Z [DEBUG] plugin.terraform-provider-auth0: [signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x181312a]
2018-11-04T17:53:00.507Z [DEBUG] plugin.terraform-provider-auth0: 
2018-11-04T17:53:00.507Z [DEBUG] plugin.terraform-provider-auth0: goroutine 42 [running]:
2018-11-04T17:53:00.507Z [DEBUG] plugin.terraform-provider-auth0: github.com/yieldr/terraform-provider-auth0/auth0.readCustomDomain(0xc000303500, 0x19980e0, 0xc000236a50, 0xc000303500, 0x0)
2018-11-04T17:53:00.507Z [DEBUG] plugin.terraform-provider-auth0:       /go/src/github.com/yieldr/terraform-provider-auth0/auth0/resource_auth0_custom_domain.go:87 +0x21a
2018-11-04T17:53:00.507Z [DEBUG] plugin.terraform-provider-auth0: github.com/yieldr/terraform-provider-auth0/vendor/github.com/hashicorp/terraform/helper/schema.(*Resource).Refresh(0xc0004295e0, 0xc000308500, 0x19980e0, 0xc000236a50, 0xc000391000, 0x10bd101, 0x187fc20)
2018-11-04T17:53:00.507Z [DEBUG] plugin.terraform-provider-auth0:       /go/src/github.com/yieldr/terraform-provider-auth0/vendor/github.com/hashicorp/terraform/helper/schema/resource.go:352 +0x160
2018-11-04T17:53:00.507Z [DEBUG] plugin.terraform-provider-auth0: github.com/yieldr/terraform-provider-auth0/vendor/github.com/hashicorp/terraform/helper/schema.(*Provider).Refresh(0xc000429960, 0xc0003084b0, 0xc000308500, 0xc0002ca800, 0x18, 0x2601440)
2018-11-04T17:53:00.507Z [DEBUG] plugin.terraform-provider-auth0:       /go/src/github.com/yieldr/terraform-provider-auth0/vendor/github.com/hashicorp/terraform/helper/schema/provider.go:308 +0x92
2018-11-04T17:53:00.507Z [DEBUG] plugin.terraform-provider-auth0: github.com/yieldr/terraform-provider-auth0/vendor/github.com/hashicorp/terraform/plugin.(*ResourceProviderServer).Refresh(0xc0004018a0, 0xc00043ae60, 0xc00043af30, 0x0, 0x0)
2018-11-04T17:53:00.507Z [DEBUG] plugin.terraform-provider-auth0:       /go/src/github.com/yieldr/terraform-provider-auth0/vendor/github.com/hashicorp/terraform/plugin/resource_provider.go:549 +0x4e
2018-11-04T17:53:00.507Z [DEBUG] plugin.terraform-provider-auth0: reflect.Value.call(0xc00007b8c0, 0xc00000d7c0, 0x13, 0x1a0dffa, 0x4, 0xc000099f18, 0x3, 0x3, 0xc000304180, 0x0, ...)
2018-11-04T17:53:00.507Z [DEBUG] plugin.terraform-provider-auth0:       /usr/local/go/src/reflect/value.go:447 +0x449
2018-11-04T17:53:00.507Z [DEBUG] plugin.terraform-provider-auth0: reflect.Value.Call(0xc00007b8c0, 0xc00000d7c0, 0x13, 0xc000314718, 0x3, 0x3, 0x0, 0x0, 0x0)
2018-11-04T17:53:00.507Z [DEBUG] plugin.terraform-provider-auth0:       /usr/local/go/src/reflect/value.go:308 +0xa4
2018-11-04T17:53:00.508Z [DEBUG] plugin.terraform-provider-auth0: net/rpc.(*service).call(0xc000436680, 0xc0003edc70, 0xc000420b88, 0xc000420ba0, 0xc00010be00, 0xc000401900, 0x187fbe0, 0xc00043ae60, 0x16, 0x187fc20, ...)
2018-11-04T17:53:00.508Z [DEBUG] plugin.terraform-provider-auth0:       /usr/local/go/src/net/rpc/server.go:384 +0x14e
2018-11-04T17:53:00.508Z [DEBUG] plugin.terraform-provider-auth0: created by net/rpc.(*Server).ServeCodec
2018-11-04T17:53:00.508Z [DEBUG] plugin.terraform-provider-auth0:       /usr/local/go/src/net/rpc/server.go:481 +0x47e
2018/11/04 17:53:00 [ERROR] root: eval: *terraform.EvalRefresh, err: auth0_client.this: unexpected EOF
2018/11/04 17:53:00 [ERROR] root: eval: *terraform.EvalRefresh, err: auth0_custom_domain.this: unexpected EOF
2018/11/04 17:53:00 [ERROR] root: eval: *terraform.EvalSequence, err: auth0_client.this: unexpected EOF
2018/11/04 17:53:00 [ERROR] root: eval: *terraform.EvalSequence, err: auth0_custom_domain.this: unexpected EOF
2018/11/04 17:53:00 [TRACE] [walkRefresh] Exiting eval tree: auth0_custom_domain.this
2018/11/04 17:53:00 [TRACE] [walkRefresh] Exiting eval tree: auth0_client.this
```

I wish I knew more Golang to raise a PR. Sorry.

--- a/auth0/resource_auth0_custom_domain.go
+++ b/auth0/resource_auth0_custom_domain.go
@@ -83,9 +83,13 @@ func readCustomDomain(d *schema.ResourceData, m interface{}) error {
 	d.Set("type", c.Type)
 	d.Set("primary", c.Primary)
 	d.Set("status", c.Status)
-	d.Set("verification", []map[string]interface{}{
-		{"methods": c.Verification.Methods},
-	})
+
+	if c.Verification != nil {
+		d.Set("verification", []map[string]interface{}{
+			{"methods": c.Verification.Methods},
+		})
+	}
+
 	return nil
 }
 

